### PR TITLE
Add osmnx-base

### PR DESCRIPTION
**Pull request by cheginit**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by cheginit** [comment]
@conda-forge-admin, please rerender

**Comment by cheginit** [comment]
Following #116, this PR adds a base output.

**Comment by cheginit** [comment]
@conda-forge-admin, please rerender

**Comment by cheginit** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml
Comment (gboeing) [inline recipe/meta.yaml:17]: Does this need to be a separate file or can you just stick the one liner here?

Or can you just stick it in line 16 above (where you had removed it), to use it across the recipe rather than defining a different build script for each output?

File: recipe/meta.yaml
Comment (gboeing) [inline recipe/meta.yaml:19]: Isn't `noarch` defined on line 15 already?

File: recipe/meta.yaml
Comment (gboeing) [inline recipe/meta.yaml]: Is this necessary if the package is pure Python? Why would we need `cross-python_` anything?

File: recipe/meta.yaml
Comment (gboeing) [inline recipe/meta.yaml]: Let's just continue using the `geopandas` dependency rather than cherry-picking sub-dependencies from `geopandas-base` (whose min values we'd need to keep track of and maintain here).

**Review by gboeing — CHANGES_REQUESTED** [review]

File: recipe/meta.yaml
Comment (cheginit) [reply recipe/meta.yaml:17]: No, it seems that for multi-output recipe, a script file needs to be used. I tried with using the command directly and it failed.

File: recipe/meta.yaml
Comment (cheginit) [reply recipe/meta.yaml:19]: It appears that both need to be defined, otherwise it will not consider the build to be `noarch`. I tried without this first but realized that it needs to be defined.

File: recipe/meta.yaml
Comment (cheginit) [reply recipe/meta.yaml]: I am not sure, I copied this line from GeoPandas recipe. Maybe @ocefpaf can explain this?

File: recipe/meta.yaml
Comment (cheginit) [reply recipe/meta.yaml]: Right, I understand your concern. The main reason that I added `geopandas` is that it adds several additional deps that I don't think are needed: `matplotlib-base`, `mapclassify`, `folium`, and `xyzservices`. They are needed mainly for the `explore` method of `geopandas`.

**Comment by cheginit** [comment]
@ocefpaf can you please check why the validation for `osmnx-base` fails at this [line](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=1112098&view=logs&j=7b6f2c87-f3a7-5133-8d84-7c03a75d9dfc&t=9eb77fd2-8ddd-5444-8fc0-71cb28dcb736&l=1478) of the log. The log in not helpful.

**Comment by cheginit** [comment]
@conda-forge-admin, please rerender

**Comment by cheginit** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml
Comment (cheginit) [reply recipe/meta.yaml]: What about keeping this for the `osmnx-base` output and add `geopandas` to the `osmnx` output? This will be used by maintainers anyway, and they can adjust the min versions in their own recipe if needed.

Comment (gboeing) [reply recipe/meta.yaml]: I think this would be an issue to raise upstream at the GeoPandas feedstock if `geopandas-base` could be made slimmer by avoiding optional dependencies only needed by the `explore` method. But for our feedstock here, I don't think we want to get into the game of cherry-picking and maintaining min version pins for upstream dependencies' dependencies.

Comment (cheginit) [reply recipe/meta.yaml]: Sounds good, I made the necessary changes.

**Comment by cheginit** [comment]
@gboeing, is the `yum_requirements.txt` file necessary? The deps of `osmnx` are not very different from `geopandas`' and they don't have it.

**Comment by gboeing** [comment]
@cheginit I'm not sure. I thought that had been autogenerated by conda-smithy at some point. But I'm away from the computer and can't check the file history right now.

**Comment by ocefpaf** [comment]
> @gboeing, is the `yum_requirements.txt` file necessary? The deps of `osmnx` are not very different from `geopandas`' and they don't have it.


It depends, some plotting tests may require it. However, it doesn't seems to be used here.

@cheginit the failure is b/c this introduces a new output and we need to validate it before merging this PR. See https://github.com/conda-forge/admin-requests/pull/1228 for an example on how to request a new output for this feedstock.

**Comment by cheginit** [comment]
@ocefpaf Ah, I see. Thanks! I just opened a [PR](https://github.com/conda-forge/admin-requests/pull/1234).

@gboeing Considering the explanation by @ocefpaf for the yum file, please let me know if you prefer to remove the file or keep it as is.

File: recipe/meta.yaml
Comment (ocefpaf) [reply recipe/meta.yaml]: You probably don't need. If I recall correctly we needed that for a workaround at some point.

File: recipe/meta.yaml
Comment (ocefpaf) [inline recipe/meta.yaml]: The name is used in the URL multiple times but this one only once. Let's just hardcode it there to make the recipe smaller and clearer.

File: recipe/meta.yaml
Comment (cheginit) [reply recipe/meta.yaml]: @ocefpaf Right, I removed it and it worked.

File: recipe/meta.yaml
Comment (ocefpaf) [inline recipe/meta.yaml]: We only need to define this if the min python is higher then the EOL. Python 3.9 is conda-forge's min now.

```suggestion
```

File: recipe/meta.yaml
Comment (gboeing) [inline recipe/meta.yaml:3]: Where is `python_min` defined now? I don't see it, but it's still referenced throughout this yaml file.

File: recipe/meta.yaml
Comment (gboeing) [reply recipe/meta.yaml:19]: Is this still the case now that the multiple outputs have been approved?

File: recipe/meta.yaml
Comment (gboeing) [inline recipe/meta.yaml:48]: We don't need `hatchling` and `pip` here anymore?

File: recipe/meta.yaml
Comment (gboeing) [reply recipe/meta.yaml:17]: Is this still the case now that the multiple outputs have been approved?

**Review by gboeing — CHANGES_REQUESTED** [review]

File: recipe/meta.yaml
Comment (gboeing) [inline recipe/meta.yaml]: I thought I left a comment on this line but I don't see it now.

Doesn't this already come in via the `pin_subpackages` below?

File: recipe/meta.yaml
Comment (cheginit) [reply recipe/meta.yaml:3]: Please see @ocefpaf's comment:

> We only need to define this if the min python is higher then the EOL. Python 3.9 is conda-forge's min now.

**Review by gboeing — CHANGES_REQUESTED** [review]

File: recipe/meta.yaml
Comment (cheginit) [reply recipe/meta.yaml:48]: No, they're needed only when building.

File: recipe/meta.yaml
Comment (cheginit) [reply recipe/meta.yaml]: I'm not sure, I left it there, since `geopandas`'s feedstock has it. I can try removing it and see if it works, if you want.

File: recipe/meta.yaml
Comment (gboeing) [reply recipe/meta.yaml:48]: But we do have them in the "host" section above for `osmnx-base`?

File: recipe/meta.yaml
Comment (gboeing) [reply recipe/meta.yaml]: Yes it looks like you should be able to remove it.

File: recipe/meta.yaml
Comment (gboeing) [reply recipe/meta.yaml:3]: Got it. I see his comment now.

**Comment by gboeing** [comment]
> Considering the explanation by @ocefpaf for the yum file, please let me know if you prefer to remove the file or keep it as is.

@cheginit sure you can remove it then if it's not needed.

File: recipe/meta.yaml
Comment (cheginit) [reply recipe/meta.yaml:48]: Although builds pass successfully without them, I just added them.

File: recipe/meta.yaml
Comment (cheginit) [reply recipe/meta.yaml:19]: Yes, the issue caused by this was different from the approval one. Now, each output has its own `noarch` block.

**Comment by cheginit** [comment]
@gboeing I applied the requested changes and all builds pass successfully.

**Review by gboeing — APPROVED** [review]

**Pull request merged**
